### PR TITLE
Route to exploration in promoted story list (Continue Playing)

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

This PR mainly changes the item click route in ContinuePlaying Screen.
Earlier it was redirecting to `ChapterList` screen but now it will directly start the exploration and when we leave this screen it should intent to `TopicActivity` and not `ContinuePlayingActivity` again.

continue_playing_routing

## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by rt4914** [comment]
@BenHenning I have not updated the test cases in this PR, because in this actually all test cases needs update but the main issue is, that as soon as the domain layer promoted list changes these test cases will again fail, so what can be the best way to test `ContinuePlayingActivity`.

**Comment by BenHenning** [comment]
@rt4914 we ideally need to set up the test data to have the desired promoted story set up regardless of how the domain layer handles this. We're a bit limited on how we do this until we finish #77. We'll need to hardcode behavior based on the domain layer, and fix the domain layer data to make it work until we finish #77.

**Comment by mschanteltc** [comment]
The flow matches what I described. LGTM.

**Comment by rt4914** [comment]
> @rt4914 we ideally need to set up the test data to have the desired promoted story set up regardless of how the domain layer handles this. We're a bit limited on how we do this until we finish #77. We'll need to hardcode behavior based on the domain layer, and fix the domain layer data to make it work until we finish #77.

Okay got it thanks. So in that case, we will need to update the domain layer to show 2 recent stories and 2 old stories which will be an ideal testing case for `HomeFragment` carousel as well as `ContinuePlaying` screen.

**Comment by nikitamarysolomanpvt** [comment]
Please assign once changes are implemented...

**Comment by rt4914** [comment]
@veena14cs @nikitamarysolomanpvt @BenHenning  This PR has become big because of following chain reaction:

1. Introduced topicId in intent to ExplorationActivity, which resulted in change in all the activities which were calling `ExplorationActivity`.
2. Because there were changes in multiple activities their test cases were also needed to be fixed which results in more files.
3. Dummy `Play Exploration` button was removed from the `HomeActivity` which resulted in creating another test activity `ExplorationTestAcvtity` which can check all the test cases related to that button.

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -26,6 +26,8 @@
       android:theme="@style/OppiaThemeWithoutActionBar"
       android:screenOrientation="portrait"
       android:windowSoftInputMode="adjustResize"/>
+    <activity
+      android:name=".testing.ExplorationTestActivity" />
```
Comment (veena14cs) [inline app/src/main/AndroidManifest.xml:30]: nit change maintain alphabetical order for activities.

File: app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt [hunk 2]
```diff
@@ -77,7 +77,7 @@ class ExplorationActivityTest {
 
   private fun createExplorationActivityIntent(explorationId: String): Intent {
     return ExplorationActivity.createExplorationActivityIntent(
-      ApplicationProvider.getApplicationContext(), explorationId
+      ApplicationProvider.getApplicationContext(), explorationId, null
```
Comment (veena14cs) [inline app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt:80]: Provide inline comment for null over here.

**Review by veena14cs — APPROVED** [review]
LGTM. Please address some nit changes.

**Review by nikitamarysolomanpvt — APPROVED** [review]
LDTM. All test cases are passing.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (rt4914) [reply app/src/main/AndroidManifest.xml:30]: Done.

File: app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt (hunk 2 above)
Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt:80]: Done.

**Comment by rt4914** [comment]
Merging this PR so that we can move ahead on testing other test cases on develop.

**Pull request merged**